Fix warning in sarray2.{h,c}

Hi everyone, I got curious as to why `clang` spits out the following warning when compiling:
```
sarray2.h:55:8: warning: variable length array folded to constant array as an extension [-Wgnu-folding-constant]
   55 |         void *data[data_size];
      |               ^
```
I then looked throughout the whole project into how and where `data_size` was being used:
```
find . -type f -exec grep -Hn "data_size" {} \;
./legacy.c:326:	size_t data_size = l->count * sizeof(struct objc_property);
./legacy.c:327:	struct objc_property_list *n = calloc(1, sizeof(struct objc_property_list) + data_size);
./sarray2.c:223:		for(uint32_t i=0 ; i<data_size ; i++)
./sarray2.h:21:static const uint32_t data_size = 256;
./sarray2.h:26:static const uint32_t data_mask = data_size - 1;
./sarray2.h:55:	void *data[data_size];
```
The `data_size` variable in `legacy.c` is a different thing with the same name, so it is irrelevant to this particular issue, but I left it in the above output for completeness. Our particular `data_size` is being folded into a constant, and since it is not marked as `volatile` or `extern`, I believe it makes more sense to convert it into a macro and get rid of the warning, especially since it isn't being used in a funky way.

The fix is trivial:
`#define DATA_SIZE 256`

Then, it's merely a matter of updating two places in `sarray2.h`, and one place in `sarray2.c`.

My overthinking brain, however, wonders why `data_size` was initially defined as a variable and not a macro. Is there something deeper going on that I haven't observed? It could be just a simple oversight, but I like to turn all my metaphorical stones over on all sides before claiming to have found a fix.

> It's not defined as a variable, it's defined as a constant.

Right, oops. My bad. After thinking about it for a minute, using an integer constant does seem like a more expressive choice than a macro.

> And it's a limitation of old versions of the C standard that integer constants can't be used as integer constant expressions. This is fixed in C23.

To test this out, I added the following lines to the CMake file:

```
set(CMAKE_C_STANDARD 23)
set(CMAKE_C_STANDARD_REQUIRED ON)
```

I'm on Fedora 42 running clang 20.1.8 (just to get that out of the way); after I setup a fresh build, `make VERBOSE=1` still produces the following output even with the `-std=gnu23` flag set:

```
[ 48%] Building C object CMakeFiles/objc.dir/sarray2.c.o
/usr/bin/clang -DCXA_ALLOCATE_EXCEPTION_SPECIFIER=noexcept -DEMBEDDED_BLOCKS_RUNTIME -DGNUSTEP -DNO_LEGACY -DOLDABI_COMPAT=1 -DTYPE_DEPENDENT_DISPATCH -D__OBJC_BOOL -D__OBJC_RUNTIME_INTERNAL__=1 -Dobjc_EXPORTS -I/home/me/Documents/libobjc2/build/objc -I/home/me/Documents/libobjc2/build/_deps/robinmap-src/include -std=gnu23 -fPIC -Xclang -fexceptions -MD -MT CMakeFiles/objc.dir/sarray2.c.o -MF CMakeFiles/objc.dir/sarray2.c.o.d -o CMakeFiles/objc.dir/sarray2.c.o -c /home/me/Documents/libobjc2/sarray2.c
In file included from /home/me/Documents/libobjc2/sarray2.c:6:
/home/me/Documents/libobjc2/sarray2.h:55:8: warning: variable length array folded to constant array as an extension [-Wgnu-folding-constant]
   55 |         void *data[data_size];
      |               ^
1 warning generated.
```

What gives? I'd like to know what you make of this (as pointless of an "issue" as it may be), if only to satisfy my intellectual curiosity :)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -258,7 +258,7 @@ else ()
 endif ()
 
 add_library(objc SHARED ${libobjc_C_SRCS} ${libobjc_ASM_SRCS} ${libobjc_OBJC_SRCS} ${libobjc_OBJCXX_SRCS} ${libobjc_ASM_OBJS})
-target_compile_options(objc PRIVATE "$<$<OR:$<COMPILE_LANGUAGE:OBJC>,$<COMPILE_LANGUAGE:OBJCXX>>:-Wno-deprecated-objc-isa-usage;-Wno-objc-root-class;-fobjc-runtime=gnustep-2.0>$<$<COMPILE_LANGUAGE:C>:-Xclang;-fexceptions>")
+target_compile_options(objc PRIVATE "$<$<OR:$<COMPILE_LANGUAGE:OBJC>,$<COMPILE_LANGUAGE:OBJCXX>>:-Wno-gnu-folding-constant;-Wno-deprecated-objc-isa-usage;-Wno-objc-root-class;-fobjc-runtime=gnustep-2.0>$<$<COMPILE_LANGUAGE:C>:-Xclang;-fexceptions;-Wno-gnu-folding-constant>")
 
 list(APPEND libobjc_CXX_SRCS ${libobjcxx_CXX_SRCS})
 target_sources(objc PRIVATE ${libobjc_CXX_SRCS})

